bin/sozo/src/ops/migration/mod.rs

```diff
@@ -83,38 +84,52 @@ pub async fn execute(
     ui.print_sub(format!("Total diffs found: {total_diffs}"));
 
     if total_diffs == 0 {
-        ui.print("\n✨ No changes to be made. Remote World is already up to date!")
-    } else {
-        // Migrate according to the diff.
-        let migration_output = apply_diff(
-            ws,
-            &target_dir,
-            diff,
-            name,
-            world_address,
-            &account,
-            Some(args.transaction),
-        )
-        .await?;
-
-        update_manifests_and_abis(
-            ws,
-            local_manifest,
-            remote_manifest,
-            &manifest_dir,
-            migration_output,
-            &chain_id,
-        )
-        .await?;
+        ui.print("\n✨ No changes to be made. Remote World is already up to date!");
+        return Ok(());
     }
 
+    let strategy = prepare_migration(&target_dir, diff, name, world_address, &ui)?;
+    let world_address = strategy.world_address().expect("world address must exist");
+
+    // Migrate according to the diff.
+    match apply_diff(ws, &account, None, &strategy).await {
+        Ok(migration_output) => {
+            update_manifests_and_abis(
+                ws,
+                local_manifest,
+                remote_manifest,
+                &manifest_dir,
+                migration_output,
+                &chain_id,
+            )
+            .await?;
+        }
+        Err(e) => {
+            update_manifests_and_abis(
+                ws,
+                local_manifest,
+                remote_manifest,
+                &manifest_dir,
+                MigrationOutput {
+                    world_address,
+                    world_tx_hash: None,
+                    world_block_number: None,
+                    success: false,
```

Comment:
Could we derive `Default` on `MigrationOutput` to have `..Default::default()` instead?